src/auth/index.ts

```diff
@@ -0,0 +1,7 @@
+import { impl } from "./implementation.js"
+
+const {  isUsernameAvailable, isUsernameValid, register } = impl
+
+export { isUsernameAvailable, isUsernameValid, register }
+export { AccountLinkingRequestor, createConsumer as createRequestor } from "./linking/consumer.js"
+export { AccountLinkingProvider, createProducer as createProvider } from "./linking/producer.js"
```

Comment:
All app owned auth functions are collected up and re-exported here. We also rename consumer and producer to requestor and provider.

Reply:
Actually, why do we rename them? Should we perhaps use `requestor` and `provider` everywhere?

Reply:
We could! The renaming is in part an artifact starting with `consumer` and `producer` and arriving at the other names later. 

But there are a couple of other considerations:
- The whitepaper uses the terms `consumer` and `producer`. Keeping these terms makes it easier to compare the implementation with the spec.
- We may re-purpose parts of the `consumer` and `producer` modules for other AWAKE purposes (account recovery, chat, etc) where the names `requestor` and `provider` may not make sense. `consumer` and `producer` are generic, architectural terms that apply broadly.

Reply:
This sounds like `consumer` and `producer` are better names. I'd vouch for using them consistently everywhere then :+1:

Reply:
The goal of renaming them is to make them more intuitive in the API. In my opinion, consumer and producer don't carry enough meaning in the context of account linking unless you are familiar with the implementation or have read the whitepaper. Provider and requestor carry more implication of their purpose.

There is absolutely a tradeoff here. Folks who are interested in reading the whitepaper to understand the implementation will need to take a first step translating the terms when reading the code. We also have a bit of cognitive overhead maintaining the code. On the other hand, folks who just want to use the API have clearer, more meaningful terms.

Reply:
I get this take but in general there isn’t a good way around this for ALL auth / identity related terms. eG IDP =
Identity Provider. 

If people are using this as a library where they aren’t needing to write code, maybe alignment with the spec version is useful. 

I’d like to see an external user say “this is confusing” first. I don’t at first read see requestor / provider being that much clearer?

the two “p” words of producer / provider seem too close especially?

Reply:
As an eyeball comparison, let's look at both options:

```typescript
const provider = await auth.createProvider({ username })
const requestor = await auth.createRequestor({ username })
```
A provider provides auth, and a requestor requests auth.

```typescript
const producer = await auth.createProducer({ username })
const consumer = await auth.createConsumer({ username })
```
A produces produces auth, and a consumer consumes auth.

Providing and requesting auth make sense intuitively, where, at first read, I question what it means to produce and consume auth. Again, this is just my intuition on it! This is highly subjective and my view isn't necessarily right.

> If people are using this as a library where they aren’t needing to write code, maybe alignment with the spec version is useful.

That's a fair point! If someone only looked at the API docs and the whitepaper, this might not be clear.

> I get this take but in general there isn’t a good way around this for ALL auth / identity related terms. eG IDP =
Identity Provider.

Even if there aren't good standard terms, it's worth thinking about what terms we can draw intuitions from with the goal of making the API self-evident to developers.

Reply:
Another thing to consider is if we do re-use portions of the consumer and producer for other use cases, we may want names that are more specific. For example, some parts of the consumer and producer will be usable in account recovery (and maybe more!).

One other way to approach this would be to differentiate consumers and producers by top level export.

Reply:
To me neither of these are clearer.

> Providing and requesting auth make sense intuitively

I don't think it does, especially as you look across language barriers and such. Needing a provider/requester or producer/consumer just doesn't feel different enough to me to warrant renaming.

And that's the challenge. We both have "correct" hypotheses, and won't know until we get feedback in the wild. Mostly, I think people will cut and paste code examples to start with.

Not a hugely important issue to me either way.

Reply:
> And that's the challenge. We both have "correct" hypotheses, and won't know until we get feedback in the wild.

Totally agreed. It's hard to gauge without hearing from users. 💯 

> Mostly, I think people will cut and paste code examples to start with.

True! Most folks won't care all that much and will directly use what we have in docs or a demo app.

Reply:
After discussing this with @therealjeffg, I think we will avoid renaming and use consumer and producer. The names provider and requestor aren't distinct enough to warrant renaming.

We did discuss contextualizing consumer and producer at the top level export by exposing them from `account` instead of `auth`. From a developer perspective, they would look like:

```typescript
import { account } from 'webnative'

const acctProducer = await account.createProducer({ username })
```

In the future, when we find other uses for AWAKE, we can export from other top level exports. For example, we might export account recovery from `recovery`.